### PR TITLE
Change class names to only 'Metasploit'

### PR DESCRIPTION
**Pull request by firefart**
See https://github.com/rapid7/metasploit-framework/issues/6389

This is a first try in implementing it, I think there is still some work to do (specs for example).

Feedback from framework pros welcome!

### CODE REVIEW

File: modules/exploits/multi/http/joomla_http_header_rce.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 require 'msf/core'
 
-class Metasploit3 < Msf::Exploit::Remote
+class Metasploit < Msf::Exploit::Remote
```
Comment (jhart-r7) [inline modules/exploits/multi/http/joomla_http_header_rce.rb]: I'm assuming you are just changing this one as an example and if all goes well someone will update all other modules?

Comment (FireFart) [inline modules/exploits/multi/http/joomla_http_header_rce.rb]: jeah changing all modules and payloads will result in a lot of conflicts so I changed only one for the initial PR and discussion

**Comment by wvu-r7** [comment]
I feel like this will break Pro.

**Comment by FireFart** [comment]
probably, but I can not test the pro version. I think we should first find a way to integrate the changes in the open source version, and then let someone look at the pro code.

**Comment by bcook-r7** [comment]
Quick test shows it works as well in pro as it does in framework, in that it causes it to reject all Metasploit3/4 modules. We will want a grace period before rejecting old-style modules outright, since probably a not-insignificant number of users have local modules that would stop loading as well.

**Comment by FireFart** [comment]
@bcook-r7 I pushed a change to allow old names too. The idea is to implement a check in msftidy for the new class name.
I did a `load_error` on the modules with an old class name, but I couldn't figure out how to display a message to the user at that point without rejecting the module. Any ideas?

**Comment by busterb** [comment]
As a quick hoist on existing modules to see how well this would work, I added this:
```
diff --git a/lib/msf/core/modules/loader/directory.rb b/lib/msf/core/modules/loader/directory.rb
index 46f182e..a6e3702 100644
--- a/lib/msf/core/modules/loader/directory.rb
+++ b/lib/msf/core/modules/loader/directory.rb
@@ -81,6 +81,7 @@ class Msf::Modules::Loader::Directory < Msf::Modules::Loader::Base
     full_path = module_path(parent_path, type, module_reference_name)

     module_content = ''
+    old_class = /Metasploit[3,4]/

     begin
       # force to read in binary mode so Pro modules won't be truncated on Windows
@@ -89,7 +90,7 @@ class Msf::Modules::Loader::Directory < Msf::Modules::Loader::Base
         # @see http://www.ruby-forum.com/topic/209005
         # @see https://github.com/ruby/ruby/blob/ruby_1_8_7/io.c#L1205
         # @see https://github.com/ruby/ruby/blob/ruby_1_9_3/io.c#L2038
-        module_content = f.read(f.stat.size)
+        module_content = f.read(f.stat.size).sub(old_class, 'Metasploit')
       end
     rescue Errno::ENOENT => error
       load_error(full_path, error)
```

That reveals a possible problem with ssh_login_pubkey that I didn't dig into:
```
[-] WARNING! The following modules could not be loaded!
[-] 	/Users/bcook/projects/metasploit-framework/modules/auxiliary/scanner/ssh/ssh_login_pubkey.rb: NameError uninitialized constant Msf::Modules::Mod617578696c696172792f7363616e6e65722f7373682f7373685f6c6f67696e5f7075626b6579::Metasploit::Framework
```

I just moved that module out of the way for now to see what effect this has on performance:
```
Before:
time ./msfconsole -qx 'exit'

real	0m6.336s
user	0m5.524s
sys	0m0.719s

After:
time ./msfconsole -qx 'exit'

real	0m4.681s
user	0m4.045s
sys	0m0.564s
```

**Comment by FireFart** [comment]
@bcook-r7 after my latest commit I am able to load `auxiliary/scanner/ssh/ssh_login_pubkey`. Looking at the module I can see no abnormal stuff causing this error.

**Comment by busterb** [comment]
Yeah, that works fine for me as well.

**Comment by FireFart** [comment]
Now the warnings for each module are also printed out and the module is loaded correctly

**Comment by FireFart** [comment]
So what do you guys think? Is this the right way to change the class names? If so I can start modifying the rspecs

**Comment by bcook-r7** [comment]
I think it's a good move. Would like some old-timers to weigh in before making the leap.

**Comment by FireFart** [comment]
Should we also rename the `~/.msf4` folder to `~/.msf` to be consistent?

**Comment by jlee-r7** [comment]
>
> Should we also rename the ~/.msf4 folder to ~/.msf to be consistent?

We've talked about this off and on forever. I think it's a good move, but it will require some communication and probably a deprecation with fallbacks for a while. Whatever we decide on that, it belongs in its own PR, not this one.

**Comment by FireFart** [comment]
@jlee-r7 @busterb @bcook-r7 rspecs are changed so I think this PR would be ready. The modules should be changed after merging with a replace otherwise this PR would create too much conflicts

**Comment by bcook-r7** [comment]
I'm landing with the warning disabled; once we update the built-in modules, we can reenable the warnings.

**Comment by bcook-r7** [comment]
the warning-disable commit: 5a0bec81cbd2d7ddfa2c3e4ad2e8a9f932232c2b

**Comment by FireFart** [comment]
awesome thanks @bcook-r7 !

**Comment by FireFart** [comment]
Class name change PR is in #6648

**Pull request merged**